src/client/components/MealsDetail.js

```diff
@@ -0,0 +1,30 @@
+import React, { useState, useEffect } from "react";
+import Meals from "../../pages/Meals";
+import MenuMeal from "./MenuMeal";
+
+function MealsDetail({ match }) {
+  const [meal, setMeal] = useState({});
+  console.log(meal, "meal");
+
+  useEffect(() => {
+    fetchMeal();
+  }, []);
+
+  const fetchMeal = async () => {
+    const fetchMeal = await fetch(`/api/meals/${match.params.id}`);

```

Comment:
since you used context.provider so you don't need to fetch the meals in each component 
you can easily import any state from provider  it like that :   const { meals,setMeals,error, userSearchInput} = useContext(Data);

Reply:
your solution is awesome, thanks!